src/Beacon.jsx

```diff
@@ -177,8 +204,15 @@ export class Beacon extends Component {
     return targetClone;
   }
 
-  renderBeacon(persistent) {
-    if (this.state.inactive || (persistent && !this.state.hash)) {
+  getHash() {
+    const { tooltipText, id } = this.props;
+    const text = typeof(tooltipText) === 'string' ? tooltipText : ReactDOMServer.renderToStaticMarkup(tooltipText);
```

Comment:
If you can suggest a more robust way to do the `renderToStaticMarkup` thing then I'm all ears. I guess even if they move it to server-only code I can still include it somehow.